app/models/concerns/rateable.rb

```diff
@@ -29,8 +29,8 @@ def update_rating_frequency(rating, diff)
           COALESCE(rating_frequencies->'#{rating}', '0')::integer + #{diff}
         )::text)
     EOF
-    self.class.where(id: id).update_all(update_query)
-    touch
+    class_name = self.class.name.to_s
+    UpdateRatingFrequencyWorker.perform_async(class_name, id, update_query)
```

Comment:
It might make more sense if the worker takes `rating` and `diff` instead of passing the SQL as a parameter